db/utils.js

```diff
@@ -123,8 +122,11 @@ const DBUtils = {
     return await this.addBreachedHash(breachName, getSha1(email));
   },
 
-  async setBreachedHashNotified(breach, subscriber) {
-    return await BreachedHash.query().patch({notified: true}).where("breach_id", "=", breach.id).andWhere("sha1_id", "=", subscriber.id);
+  async setBreachedHashNotified(breach, email) {
+    return await breach
+      .$relatedQuery("email_hashes")
+      .where("sha1", getSha1(email))
+      .patch({ notified: true });
```

Comment:
This is how I first tried to do this, but it doesn't actually work. ...
```
blurts=# select * from breached_hashes;
 id | sha1_id | breach_id | notified
----+---------+-----------+----------
  1 |       1 |         1 | f
  2 |       3 |         1 | f
  3 |       2 |         1 | f
  4 |       3 |         2 | f
  5 |       2 |         2 | f
  6 |       4 |         2 | f
  7 |       4 |         3 | f
  9 |       1 |         3 | f
  8 |       2 |         3 | f
(9 rows)
```
Then I ran it: 
```
groovetop:blurts-server lcrouch$ node scripts/notify-subscribers-of-breach.js AllMusic
Attempting to get SMTP credentials from environment...
Found breach:  Breach {
  id: 3,
  name: 'AllMusic',
  meta:
   { date: '6 December 2015',
     dataClasses: 'Email addresses, IP addresses, Passwords, Usernames, Website activity',
     acCount: '1,436,486' } }
Found 1 un-notified subscribers in the breach
Sending email to subscriber ID: 2 ...
DONE sending email.
Setting notified=true ...
DONE setting notified=true
```
But the database table still has `false`/`f` in the `notified` column:
```
blurts=# select * from breached_hashes;
 id | sha1_id | breach_id | notified
----+---------+-----------+----------
  1 |       1 |         1 | f
  2 |       3 |         1 | f
  3 |       2 |         1 | f
  4 |       3 |         2 | f
  5 |       2 |         2 | f
  6 |       4 |         2 | f
  7 |       4 |         3 | f
  9 |       1 |         3 | f
  8 |       2 |         3 | f
(9 rows)
```

That's why I resorted to using an explicit `BreachedHash` model. It did the trick.